examples/kubectl.rs

```diff
@@ -0,0 +1,169 @@
+//! This is a simple imitation of the basic functionality of kubectl
+//! Supports kubectl {get, delete, watch} <resource> [name] (name optional) with labels and namespace selectors
+use anyhow::{bail, Result};
+use clap::{arg, command};
+use either::Either;
+use futures::{StreamExt, TryStreamExt};
+use k8s_openapi::chrono::{Duration, Utc};
+use kube::{
+    api::{Api, DynamicObject, ListParams, ObjectMeta, Resource, ResourceExt},
+    discovery::{ApiCapabilities, ApiResource, Discovery, Scope},
+    runtime::{
+        utils::try_flatten_applied,
+        wait::{await_condition, conditions::is_deleted},
+        watcher,
+    },
+    Client,
+};
+use log::info;
+use std::{collections::BTreeMap, env};
+
+type KindMap = BTreeMap<String, (ApiResource, ApiCapabilities)>;
+async fn discover_valid_inputs(client: Client) -> Result<KindMap> {
+    let mut kinds = BTreeMap::new(); // valid inputs (keys are plural or kind)
+    let discovery = Discovery::new(client).run().await?;
```

Comment:
Not specific to this, but looks like `Discovery::run` fails if *any* aggregated service is unavailable, even if it's something relatively inconsequential like metrics.

Kubectl seems to be robust against this.

Reply:
`kubectl` creates a cache of the discovery results in `~/.kube/cache`. i didn't want to overdo this example though.

Reply:
Yeah, but even then I think kubectl will just log failures and keep going with the rest.

Reply:
oh, i misunderstood. you're not talking about general failures, but aggregated ones going out to separate services. that might be something we could tweak in the discovery interface in the future, yeah.

Reply:
Yeah, exactly.